speaker/main.go

```diff
@@ -327,24 +334,32 @@ func (c *controller) deleteBalancer(l log.Logger, name, reason string) k8s.SyncS
 		return k8s.SyncStateError
 	}
 
-	announcing.Delete(prometheus.Labels{
-		"protocol": string(proto),
-		"service":  name,
-		"node":     c.myNode,
-		"ip":       c.svcIP[name].String(),
-	})
+	for _, ip := range c.svcIPs[name] {
+		announcing.Delete(prometheus.Labels{
+			"protocol": string(proto),
+			"service":  name,
+			"node":     c.myNode,
+			"ips":      ip.String(),
+		})
+	}
 	delete(c.announced, name)
-	delete(c.svcIP, name)
+	delete(c.svcIPs, name)
 
-	level.Info(l).Log("event", "serviceWithdrawn", "ip", c.svcIP[name], "reason", reason, "msg", "withdrawing service announcement")
+	level.Info(l).Log("event", "serviceWithdrawn", "ip", c.svcIPs[name], "reason", reason, "msg", "withdrawing service announcement")
 
 	return k8s.SyncStateSuccess
 }
 
-func poolFor(pools map[string]*config.Pool, ip net.IP) string {
+func poolFor(pools map[string]*config.Pool, ips []net.IP) string {
+	found := 0
```

Comment:
You need to reset found in the beginning of the loop, otherwise you may risk to take two ips from two different pools.